enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+---
+title: upgrade-without-registry
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+creation-date: 2023-06-29
+last-updated: 2023-07-26
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Upgrade without registry
+
+## Summary
+
+Provide an automated mechanism to upgrade a cluster without requiring an image
+registry server.
+
+## Motivation
+
+All these stories are in the context of disconnected clusters with limited
+resources, both in the cluster itself and in the surrounding environment:
+
+- The cluster is not connected to the Internet, or the bandwidth is very limited.
+- It isn't possible to bring up additional machines, even temporarily.
+- The resources of the cluster are limited, in particular in terms of CPU and memory.
+- The technicians responsible for performing the upgrade have no OpenShift knowledge.
+
+These clusters are usually installed at the customer site by a partner engineer
+collaborating with customer technicians.
+
+Eventually, the cluster will need to be upgraded, and then the technicians will
+need tools that make the process as simple as possible, ideally requiring no
+OpenShift knowledge.
+
+### User Stories
+
+#### Prepare an upgrade bundle
```

Comment:
If users are preparing their own bundles, we have a risk of them attempting to upgrade in a way that is not supported (a missing edge, skipping too many releases, etc.). How will we handle that?

Reply:
The idea is that the partner has a release team that creates an upgrade package, tests it, and passes it to the factory.  If an invalid upgrade is attempted anyway (e.g., if the technician tries to skip releases), then the upgrade shouldn't start.  I assume this is already taken care of today?

Reply:
I think we can mitigate this issue adding to the bundle metadata information about the requirements for the upgrade. For example, we can have the list of versions that this upgrade is applicable to. The `oc adm upgrade create bundle` command would obtain that information from Cincinatti and would include it into the bundle. The cluster version operator would check those requirements, similar to how it checks available disk space. Would that address your concern?

Reply:
> For example, we can have the list of versions that this upgrade is applicable to. The oc adm upgrade create bundle command would obtain that information from Cincinatti and would include it into the bundle.

This should be a requisite for this feature. However this is not full proof as we can declare an update path as conditional update if we find upgrade blockers post release. In that case the partner needs to rebuild the bundle image.